EUR-Lex.js

```diff
@@ -37,42 +37,74 @@
 
 
 // attr()/text() v2
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+function attr(docOrElem, selector, attr, index) {
+	var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector); return elem ? elem.getAttribute(attr) : null;
+}
 
+function getQueryParam(url, param) {
+	const queryString = url.split("?")[1];
+	const vars = queryString.split("&");
+	for (var i = 0; i < vars.length; i++) {
+		const pair = vars[i].split("=");
+		if (pair[0] == param) {
+			return pair[1];
+		}
+	}
+	return (false);
+}
 
 // the eli resource types are described at:
 // http://publications.europa.eu/mdr/resource/authority/resource-type/html/resourcetypes-eng.html
 var typeMapping = {
-	'DIR': 'bill', // directive
-	'REG': 'statute', // regulation
-	'DEC': 'statute', // decision
-	'RECO': 'report', // recommodation
-	'OPI': 'report' // opinion
+	DIR: 'bill', // directive
+	REG: 'statute', // regulation
+	DEC: 'statute', // decision
+	RECO: 'report', // recommendation
+	OPI: 'report', // opinion
+	CASE: 'case', // case
+	CASE_LAW: 'case', // case law
+	OPIN_AG: 'case', // advocate general's opinion
+	OPIN_CASE: 'case', // advocate general's opinion
+	VIEW_AG: 'case', // advocate general's opinion
 };
 
 
 function detectWeb(doc, url) {
-	var eliTypeURI = attr(doc, 'meta[property="eli:type_document"]', 'resource');
+	const celex = getQueryParam(url, 'uri');
+	if (celex) {
+		const sector = celex.slice(6, 7);
+		if (sector === '6') { // caselaw
```

Comment:
Simplify
```js
if (celex && celex.charAt(6) === '6') {
```
Can you then also explain more in the comment what is the reason for this?